components/entities/api/src/main/java/org/phenotips/entities/PrimaryEntity.java

```diff
@@ -53,7 +55,14 @@
      *
      * @return a valid document reference
      */
-    DocumentReference getDocument();
+    DocumentReference getDocumentReference();
+
+    /**
+     * Returns the document where the entity is stored.
+     *
+     * @return a valid document reference
+     */
+    XWikiDocument getDocument();
```

Comment:
To clarify, in some cases compatibility _cannot_ easily be fixed. For example, we have provided JARs to a client a few months ago containing a module that has a controller based on the old method. We do not support their servers or provide ongoing updates to their modules. 

If we change the return type of the existing method:
They will see a `NoSuchMethodException` the next time they upgrade PT and will likely ask us why it broke and what needs to be done to fix it, at which time we'll need to step in and help them.

If we create a new method:
None of the above will happen. The module will continue to work as expected.